lib/winapi/sync.c

```diff
@@ -580,3 +580,80 @@ BOOL ReleaseMutex (HANDLE hMutex)
     SetLastError(RtlNtStatusToDosError(status));
     return FALSE;
 }
+
+HANDLE CreateEventA (LPSECURITY_ATTRIBUTES lpEventAttributes, BOOL bManualReset, BOOL bInitialState, LPCSTR lpName)
+{
+    NTSTATUS status;
+    HANDLE handle;
+    ANSI_STRING obj_name;
+    OBJECT_ATTRIBUTES obj_attributes;
+    POBJECT_ATTRIBUTES obj_attributes_ptr;
+    EVENT_TYPE event;
+
+    if (lpName) {
+        RtlInitAnsiString(&obj_name, lpName);
+        InitializeObjectAttributes(&obj_attributes, &obj_name, OBJ_OPENIF, ObWin32NamedObjectsDirectory(), NULL);
+        obj_attributes_ptr = &obj_attributes;
+    } else {
+        obj_attributes_ptr = NULL;
+    }
+
+    if (bManualReset) {
+        event = NotificationEvent;
+    } else {
+        event = SynchronizationEvent;
+    }
+
+    status = NtCreateEvent(&handle, obj_attributes_ptr, event, bInitialState);
+    if (!NT_SUCCESS(status)) {
+        SetLastError(RtlNtStatusToDosError(status));
+        return NULL;
+    }
+
+    if (status == STATUS_OBJECT_NAME_EXISTS) {
+        SetLastError(ERROR_ALREADY_EXISTS);
+    } else {
+        SetLastError(0);
+    }
+
+    return handle;
+}
+
+BOOL SetEvent (HANDLE hEvent)
+{
+    NTSTATUS status;
+
+    status = NtSetEvent(hEvent,NULL);
```

Comment:
Question: Do need to keep variable declaration and the setter couple lines down separate? Seems like we can migrate into one line. (the same for below)

If maintainers say they shouldn't, then dismiss this question.

Reply:
Hmm... I recently notice `ReleaseMutex` and `ReleaseSemaphore` functions does the same thing little bit up higher. Please dismiss this conversation.